assets/sharedClientFunctions.js

```diff
@@ -47,7 +48,7 @@ module.exports = {
         return "Make sure you enter all the information";
       }
       if (((ref.type === "giveaway" || ref.type === "eggcheck") && !regexpGive.test(url)) ||
-        (ref.type !== "giveaway" && ref.type !== "misc" && ref.type !== "eggcheck" && !regexp.test(url)) ||
+        (ref.type !== "giveaway" && ref.type !== "misc" && ref.type !== "eggcheck" && (!regexp.test(url) && !regexpMobile.test(url))) ||
```

Comment:
If we have a permalink that matches `regexp` but not `regexpMobile` or vice versa (which will I guess always be the case, since users will use one format or the other), I think having the parentheses would cause the check to fail and we'd get the error message.

```suggestion
        (ref.type !== "giveaway" && ref.type !== "misc" && ref.type !== "eggcheck" && !regexp.test(url) && !regexpMobile.test(url)) ||
```

Reply:
(Not your fault, but this logic is hard to read and maybe instead of one long thing of a bunch of `||`'s, we could consider extracting these checks out to helper functions like `isValidGiveawayLink` or `isValidTradeLink`)

Reply:
With ANDs, either will work, but cleaner to remove the parentheses as per your suggestion. I grouped them for readability, that if it doesn't match BOTH of those regex, it will fail

1 and 1 and 1 and (1 and 1) == true
1 and 1 and 1 and 1 and 1 == true
1 and 1 and 1 and (0 and 1) == false
1 and 1 and 1 and 0 and 1 == false